sql/parser/builtins.go

```diff
@@ -443,6 +425,219 @@ var builtins = map[string][]builtin{
+			return DFloat(math.Atan2(x, y)), nil
+		}),
+	},
+
+	"ceil":    {ceilImpl},
+	"ceiling": {ceilImpl},
+
+	"cos": {
+		floatBuiltin1(func(x float64) (Datum, error) {
+			return DFloat(math.Cos(x)), nil
+		}),
+	},
+
+	"degrees": {
+		floatBuiltin1(func(x float64) (Datum, error) {
+			return DFloat(180.0 * x / math.Pi), nil
+		}),
+	},
+
+	"div": {
+		floatBuiltin2(func(x, y float64) (Datum, error) {
+			return DFloat(x / y), nil
+		}),
+	},
+
+	"exp": {
+		floatBuiltin1(func(x float64) (Datum, error) {
+			return DFloat(math.Exp(x)), nil
+		}),
+	},
+
+	"floor": {
+		floatBuiltin1(func(x float64) (Datum, error) {
+			return DFloat(math.Floor(x)), nil
+		}),
+	},
+
+	"ln": {
+		floatBuiltin1(func(x float64) (Datum, error) {
+			return DFloat(math.Log(x)), nil
+		}),
+	},
+
+	"log": {
+		floatBuiltin1(func(x float64) (Datum, error) {
+			return DFloat(math.Log10(x)), nil
+		}),
+	},
+
+	"mod": {
+		floatBuiltin2(func(x, y float64) (Datum, error) {
+			return DFloat(math.Mod(x, y)), nil
+		}),
+		builtin{
+			returnType: DummyInt,
+			types:      typeList{intType, intType},
+			fn: func(args DTuple) (Datum, error) {
+				y := args[1].(DInt)
+				if y == 0 {
+					return DNull, errDivisionByZero
+				}
+				x := args[0].(DInt)
+				return DInt(x - (x/y)*y), nil
+			},
+		},
+	},
+
+	"pi": {
+		builtin{
+			returnType: DummyFloat,
+			types:      typeList{},
+			fn: func(args DTuple) (Datum, error) {
+				return DFloat(math.Pi), nil
+			},
+		},
+	},
+
+	"pow":   {powImpl},
+	"power": {powImpl},
+
+	"radians": {
+		floatBuiltin1(func(x float64) (Datum, error) {
+			return DFloat(x * math.Pi / 180.0), nil
+		}),
+	},
+
+	// TODO(thschroeter): should we support round for our floatType?
+	// or remove the implementation below?
+	//
+	// This implementation is not compatible with Postgres for
+	// rounding to a negative number of digits. We return an error,
+	// Postgres rounds digits in front of the decimal point. E.g.
+	// round(42, -1) is 40 in Postgres. Should we support this too?
+	//
+	// Postgres supports round only for their `numeric` type. That
+	// is a base-10k floating point decimal, where decimal rounding
+	// is more efficient compared to our base-2 float.
+	//
+	// The naive implementation below is not efficient since it's
+	// going through an intermediate string. It seems possible
+	// to provide a more efficient implementation that operates
+	// floats without allocating an intermediate string to represent
+	// the number as decimal before rounding. Do we need that?
+	// How is round used? In case it's mostly used to present a
+	// handful of results in a readable way it doesn't need to be
+	// performant.
+	"round": {
+		builtin{
+			returnType: DummyFloat,
+			types:      typeList{floatType, intType},
+			fn: func(args DTuple) (Datum, error) {
+				n := args[1].(DInt)
+				if n < 0 {
+					return DNull, errRoundNegativeDigits
+				}
+				x := args[0].(DFloat)
+				f := fmt.Sprintf("%%.%df", n)
```

Comment:
This could use `strconv.FormatFloat(x, 'f', n, 64)` instead of the double `Sprintf`(if it doesn't get removed completely). We also shouldn't ignore the error from `ParseFloat` below even though it really shouldn't be possible.